src/main/java/io/swagger/api/media/MediaApiController.java

```diff
@@ -40,23 +47,54 @@ public MediaApiController(ObjectMapper objectMapper, HttpServletRequest request,
     }
 
     public ResponseEntity<SearchData> mediaGet(@NotNull @Parameter(in = ParameterIn.QUERY, description = "" ,required=true,schema=@Schema()) @Valid @RequestParam(value = "query", required = true) String query) {
-        SearchData searchData = mediaService.getMediaByTitle(query);
-        return ResponseEntity.ok(searchData);
+        try {
+            SearchData searchData = mediaService.getMediaByTitle(query);
+            searchData.add(linkTo(methodOn(MediaApiController.class).mediaGet(query)).withSelfRel());
+            searchData.add(linkTo((WatchlistApiController.class)).withRel("create-watchlist"));
+            searchData.add(linkTo(methodOn(WatchlistApiController.class).watchlistWatchlistIdMediaPost(UUID.randomUUID(), new MediaItem())).withRel("add-to-watchlist"));
+            for (SearchResult result : searchData.getResults()) {
+                searchData.add(linkTo(methodOn(MediaApiController.class).mediaMediaIdGet(result.getId())).withRel("get-media-metadata"));
+            }
+            return ResponseEntity.ok(searchData);
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new ApiException(HttpStatus.INTERNAL_SERVER_ERROR.value(), String.format("Error getting media with query %s",query),e);
```

Comment:
We could probably add this code in the ControllerAdvice class so that all uncaught exceptions are translated into this format. For the sake of the project, I don't think it matters too much either way.